specs/core/0_beacon-chain.md

```diff
@@ -503,12 +512,74 @@ def int_sqrt(n: int) -> int:
     return x
 ```
 
+### PoW chain contract
+
+The beacon chain is initialized when a condition is met inside a contract on the existing PoW chain. This contract's code in Vyper is as follows:
+
+```python
+HashChainValue: event({prev_tip: bytes32, data: bytes[2048], value: wei_value, time: timestamp, total_deposit_count: int128})
+ChainStart: event({hash_chain_tip: bytes32, time: timestamp})
+
+receipt_tree: bytes32[int128]
+total_deposit_count: int128
+
+@payable
+@public
+def deposit(data: bytes[2048]):
+    log.HashChainValue(self.receipt_tree[1], data, msg.value, block.timestamp, self.total_deposit_count)
+    index:int128 = self.total_deposit_count + 2**32
```

Comment:
We changed the index to `int64` in the Specials but the index can still exceed `int64` here

Reply:
It can't, because that would require 2**64 - 2**32 deposits to happen.